src/Storages/MergeTree/MergeTreeDataPartDistinctPartitionExpressionCloner.cpp

```diff
@@ -0,0 +1,81 @@
+#include "MergeTreeDataPartDistinctPartitionExpressionCloner.h"
+#include <Common/escapeForFileName.h>
+#include <Interpreters/MergeTreeTransaction.h>
+
+namespace DB
+{
+
+    MergeTreeDataPartDistinctPartitionExpressionCloner::MergeTreeDataPartDistinctPartitionExpressionCloner(
+        MergeTreeData * merge_tree_data_,
+        const DataPartPtr & src_part_,
+        const MergeTreePartInfo & dst_part_info_,
+        const String & tmp_part_prefix_,
+        const MergeTreeTransactionPtr & txn_,
+        const MergeTreePartition & new_partition_,
+        const IMergeTreeDataPart::MinMaxIndex & new_min_max_index_
+    )
+    : MergeTreeDataPartCloner(merge_tree_data_, src_part_, merge_tree_data_->getInMemoryMetadataPtr(),
+                              dst_part_info_, tmp_part_prefix_, txn_, false, {}, false),
+        new_partition(new_partition_), new_min_max_index(new_min_max_index_)
+    {}
+
+    void MergeTreeDataPartDistinctPartitionExpressionCloner::deleteMinMaxFiles(
+        const DataPartStorageBuilderPtr & storage_builder
+    ) const
+    {
+        for (const auto & column : src_part->getColumns())
+        {
+            auto file = "minmax_" + escapeForFileName(column.name) + ".idx";
+            storage_builder->removeFile(file);
+        }
+    }
+
+    void MergeTreeDataPartDistinctPartitionExpressionCloner::updateMinMaxFiles(
+        const MutableDataPartPtr & dst_part,
+        const DataPartStorageBuilderPtr & storage_builder
+    ) const
+    {
+        deleteMinMaxFiles(storage_builder);
```

Comment:
Not sure that such number of helpers is required, it makes it harder to be in a context (at least to me)

Reply:
What d you mean by helpers? Member methods? If so, I think they help a lot when reading the code, specially when they have good & high level naming that actually describe what they are doing.

Reply:
I think that it is easier for the reader to keep everything in one method (`MergeTreeDataPartDistinctPartitionExpressionCloner::finalizePart`), but let's leave it as-is, and leave the final word for the one who will merge it.

Reply:
> I think that it is easier for the reader to keep everything in one method (`MergeTreeDataPartDistinctPartitionExpressionCloner::finalizePart`), but let's leave it as-is, and leave the final word for the one who will merge it.

Who'll do the final review & merge this?

Reply:
I don't know, but you can either ping someone (but seems that you already did this), or wait (I guess all from the team receives notifications, but maybe just overloaded).